airbyte-db/lib/configs_db_schema.md

```diff
@@ -0,0 +1,359 @@
+  - Column names are in snake case.
+- Use ANSI SQL as much as possible so that we are not locked in Postgres.
+  - Each table has an `id` column as the primary key. Its type is `nchar(36)`. The `UUID` type is not used because UUID is not a standard SQL type.
+  - String columns are `varchar(255)` by default.
+  - Enum type is not used, because it is not a standard SQL type. It is replaced with `varchar(50)`.
+    - Also modifying enum columns in Postgres requires a migration, and there is no easy way to check if an enum already exists.
+    - We already maintain enums in Java. There is no need to go through all the trouble to do that in the database again.
+- Assign default column values whenever possible.
+- Foreign keys will be added to relevant tables.
+  - The major benefit is that jOOQ can leverage that to generate convenient helper methods to create and query associated records.
+  - The downside is that config import will be tricky. But the config import system will be broken as we add more Flyway migrations, and needs to be redesigned anyway.
+  - Foreign key columns are named as `<parent_table_name>_id`. They are always indexed.
+  - If the referenced table is polymorphic, the foreign key is broken down into two columns: `owner_type` and `owner_id`.
+- For complex properties, the properties will be expanded to its own table.
+  - The property table is named as `<parent_table_name>_<property_name>`. For example, `WORKSPACE_NOTIFICATION` specifies the notifications for a `WORKSPACE`.
+- Use Table-per-Type to implement table inheritance.
+  - Comparison with other approaches: https://stackoverflow.com/a/3579462
+  - There is a `<base_table_name>_type` column in the base table so that it is easy to know which child table to query. This column is indexed.
+  - The child table has 1:1 mapping with the base table. This is enforced by a unique index on the foreign key in the child table that references the base table.
+- Each table has two timestamp columns: `created_at` and `updated_at`.
+- Index
+  - Single column index is marked with `index`.
+  - Composite column index is marked with `composite index` and specified after column definition.
+
+## Open Questions
+- Should we keep the `standard` prefix in the table names?
+  - Currently, it is removed.
+  - Will we ever need non-standard models like custom workspace or custom connector definition?
+- Most source and destination tables have exactly the same columns. For example, `SOURCE_DEFINITION` and `DESTINATION_DEFINITION`, `SOURCE_OAUTH_PARAM` and `DESTINATION_OAUTH_PARAM`. Should we merge these tables, and add a `type` column to distinguish the rows?
+  - Things to consider
+    - Dry-ness: one table is better
+    - Query complexity: same
+    - Add new columns
+      - Separate tables: need to add new columns to both tables.
+      - One table: add new columns to one table.
+      - One table is better.
+  - Do we expect the source and definition models to diverge in the future?
```

Comment:
I think it is possible. If you look at the `ConnectorSpecification` we have divergence where we have one model but some of the configuration is just for sources and some is just for destinations. In that case, since it is community developer-facing, I wish it was more clearly delineated.

That being said, I think I agree with your conclusion. Even if they do diverge a bit there are easy ways to handle it.